src/controllers/alphabetical-search/search.controller.ts

```diff
@@ -23,97 +23,91 @@ const route = async (req: Request, res: Response) => {
-                        nearestClass = "nearest";
-                        noNearestMatch = false;
-                    }
-                }
-
-                const sanitisedCorporateName = escape(result?.items.corporate_name);
-
-                return [
-                    {
-                        classes: nearestClass,
-                        html: `<a href="${result.links.self}">${sanitisedCorporateName}</a>`
-                    },
-                    {
-                        text: result?.items.company_number
-                    },
-                    {
-                        text: capitalisedStatus
-                    }
-                ];
-            });
-
-            if (req.query.originalCompanyNumber) {
-                originalCompanyNumber = req.query.originalCompanyNumber;
-            }
-
-            showPrevLink = showLinks[0];
-            showNextLink = showLinks[1];
-
-            prevUrlSearchTerm = companyResource.results[0]?.items.ordered_alpha_key
-                ? companyResource.results[0]?.items.ordered_alpha_key : encodeURIComponent(companyResource.results[0].items.corporate_name);
-
-            nextUrlSearchTerm = companyResource.results[lastIndexPosition]?.items.ordered_alpha_key
-                ? companyResource.results[lastIndexPosition]?.items.ordered_alpha_key : encodeURIComponent(companyResource.results[lastIndexPosition].items.corporate_name);
-
-            previousUrl = "get-results?companyName=" + prevUrlSearchTerm + "&originalCompanyNumber=" + originalCompanyNumber;
-            nextUrl = "get-results?companyName=" + nextUrlSearchTerm + "&originalCompanyNumber=" + originalCompanyNumber;
-        } catch (err) {
-            searchResults = [];
-            logger.error(`${err}`);
-        }
-        res.render(templatePaths.ALPHABETICAL_SEARCH_RESULTS, {
-            searchResults, previousUrl, nextUrl, showPrevLink, showNextLink, searchTerm: companyName, templateName: templatePaths.ALPHABETICAL_SEARCH_RESULTS
+        const companyNameRequestParam = req.query.companyName as string;
+        const searchBefore = req.query.searchBefore as string || null;
+        const searchAfter = req.query.searchAfter as string || null;
+        const size = Number(req.query.size) || null;
+        const companyName = companyNameRequestParam;
+        const encodedCompanyName = encodeURIComponent(companyName);
+
+        const { companyResource, searchResults } = await getSearchResults(encodedCompanyName, cookies, searchBefore, searchAfter, size);
+
+        const { results } = companyResource;
+
+        const searchBeforeAlphaKey = results[0]?.items?.ordered_alpha_key_with_id;
+        const searchAfterAlphaKey = results[results.length - 1]?.items?.ordered_alpha_key_with_id;
+        const previousUrl = searchBeforeAlphaKey ? `get-results?companyName=${encodedCompanyName}&searchBefore=${encodeURIComponent(searchBeforeAlphaKey)}` : "";
```

Comment:
We will need to add &size=40 to  the previous and next urls as when you select previous or next you will only get 20 results on that page